spyder/plugins/ipythonconsole/plugin.py

```diff
@@ -559,7 +559,12 @@ def on_close(self, cancelable=False):
         return self.get_widget().close_all_clients()
 
     def on_mainwindow_visible(self):
-        self.create_new_client(give_focus=False)
+        cli_options = self.get_command_line_options()
+        connection_file = cli_options.connection_file
+        if connection_file is not None:
+            self.create_client_for_kernel(connection_file, give_focus=False)
```

Comment:
```suggestion
            self.create_client_for_kernel(
                self.get_widget().find_connection_file(connection_file),
                give_focus=False,
            )
```

We need to use the `find_connection_file` method from `IPythonConsoleWidget` to get the absolute path to the file in the place where it's saved by Jupyter-client.